theseus/core/robust_cost_function.py

```diff
@@ -165,3 +165,65 @@ def _supports_masking(self) -> bool:
     def _supports_masking(self, val: bool):
         self.cost_function._supports_masking = val
         self.__supports_masking__ = val
+
+class GNCRobustCostFunction(RobustCostFunction):
```

Comment:
Looks like the only difference between this class and base `RobustCostFunction` is that we pass the GNC control parameter to the loss' methods. Is this correct?

If so, it would be preferable to reuse the code of the base class and only change how the loss function is called. For example:

* Add a dictionary member `self.loss_kwargs = {}` in the base class, and add the extra params in GNC class.
* Add an abstract method like `self._evaluate_loss()` and override in GNC class. 

Any thoughts?

Reply:
The additional gnc control parameter is used in `loss.evaluate` (weighted_error) and `loss.linearize` (weighted_jacobians_error); are we OK with only overriding these 2 in `GNCRobustCostFunction` class?

Reply:
I like overriding in subclass more than adding a branch in base class.

Reply:
Sounds good! I will push a revision in 1 hour.